ui/widgets/nuxeo-html-editor.js

```diff
@@ -182,7 +182,7 @@ import './quill/quill-snow.js';
     _valueChanged() {
       if (this._editor && !this._internalChange) {
         const delta = this._editor.clipboard.convert({ html: this.value });
-        this._editor.setContents(delta, Quill.sources.USER);
+        this._editor.setContents(delta, this.readOnly ? Quill.sources.SILENT : Quill.sources.USER);
```

Comment:
Just out of curiosity, what does this `Quill.sources.SILENT` does? It is like setting the quill editor in read only?

Reply:
There are three types of sources: `USER`, `API` and `SILENT`. According to the [docs](https://quilljs.com/docs/api/#setcontents), setContent (and most other methods), won't work if source is `USER` and the editor is in read-only mode. I was not sure if we should use `API` or `SILENT` here (as both work), but went with `SILENT` for consistency with the next line of code.